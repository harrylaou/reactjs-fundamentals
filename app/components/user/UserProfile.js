import React, {Component} from 'react'
import {
  Card, CardActions, CardHeader,
  CardMedia, CardTitle
} from 'material-ui/Card'

const UserProfile = (props) => {
  let user = props.user
  let username = props.username
  if (user && user.username != username) {
    this.fetchUser(username)
  }
  let fullname = ''
  let email = ''
  let gender = ''

  if (user){
    fullname = `${user.name.title} ${user.name.first} ${user.name.last}`
    email = user.email
    gender = user.gender
  }

  return (
    <Card>
      <CardHeader
        title={fullname}
        subtitle={username}
        avatar={`/images/${username}_sm.jpg`}
      />
      <CardMedia
        overlay={<CardTitle title={email} subtitle={gender} />}
      >
        <img src={`/images/${username}_lg.jpg`} />
      </CardMedia>
      <CardActions />
    </Card>
  )
}

UserProfile.propTypes = {
  user: React.PropTypes.array,
  username: React.PropTypes.string
}

export default UserProfile
