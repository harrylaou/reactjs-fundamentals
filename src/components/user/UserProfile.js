import React, { Component } from 'react'
import {
  Card, CardActions, CardHeader,
  CardMedia, CardTitle
} from 'material-ui/Card'

const UserProfile = ({ user, username }) => {
  let fullname = ''
  let email = ''
  let gender = ''

  if (user){
    fullname = `${user.name.title} ${user.name.first} ${user.name.last}`
    email = user.email
    gender = user.gender
  }

  return (
    <Card>
      <CardHeader
        title={fullname}
        subtitle={username}
        avatar={`/images/${username}_sm.jpg`}
      />
      <CardMedia
        overlay={<CardTitle title={email} subtitle={gender} />}
      >
        <img src={`/images/${username}_lg.jpg`} />
      </CardMedia>
      <CardActions />
    </Card>
  )
}

UserProfile.propTypes = {
  user: React.PropTypes.object,
  username: React.PropTypes.string
}

export default UserProfile
