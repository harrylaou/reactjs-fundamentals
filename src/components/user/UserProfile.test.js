import React from 'react'
import { shallow } from 'enzyme'
import { shallowToJson } from 'enzyme-to-json'
import UserProfile from './UserProfile'

describe('<UserProfile />', () => {
  it('should render with default props', () => {

    const wrapper = shallow(
      <UserProfile />
    )
    expect(shallowToJson(wrapper.dive())).toMatchSnapshot()
  })
})