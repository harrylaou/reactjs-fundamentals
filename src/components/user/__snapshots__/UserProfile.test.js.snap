// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<UserProfile /> should render with default props 1`] = `
<Paper
  circle={false}
  rounded={true}
  style={
    Object {
      "zIndex": 1,
    }
  }
  transitionEnabled={true}
  zDepth={1}
>
  <div
    style={
      Object {
        "paddingBottom": 0,
      }
    }
  >
    <CardHeader
      avatar="/images/undefined_sm.jpg"
      title=""
    />
    <CardMedia
      overlay={
        <CardTitle
          subtitle=""
          title=""
        />
      }
    >
      <img
        src="/images/undefined_lg.jpg"
      />
    </CardMedia>
    <CardActions />
  </div>
</Paper>
`;
